netbout-web/src/test/java/com/netbout/email/EmAliasTest.java

```diff
@@ -59,19 +65,57 @@ public void sendsEmail() throws Exception {
         final Alias alias = new EmAlias(base.randomAlias(), postman);
         final Bout bout = alias.inbox().bout(alias.inbox().start());
         bout.friends().invite(base.randomAlias().name());
-        bout.messages().post("how are you?");
+        bout.messages().post(EmAliasTest.EMAIL_BODY);
         final ArgumentCaptor<Envelope> captor =
             ArgumentCaptor.forClass(Envelope.class);
         Mockito.verify(postman).send(captor.capture());
         final Message msg = captor.getValue().unwrap();
         MatcherAssert.assertThat(msg.getFrom().length, Matchers.is(1));
+        final InternetAddress from = (InternetAddress) msg.getFrom()[0];
+        MatcherAssert.assertThat(
+            from.getPersonal(),
+            Matchers.is(alias.name())
+        );
+        MatcherAssert.assertThat(
+            from.getAddress(),
+            Matchers.is("no-reply@netbout.com")
+        );
         MatcherAssert.assertThat(msg.getAllRecipients().length, Matchers.is(1));
         MatcherAssert.assertThat(msg.getSubject(), Matchers.startsWith("#1: "));
         final ByteArrayOutputStream baos = new ByteArrayOutputStream();
         MimeMultipart.class.cast(msg.getContent()).writeTo(baos);
         MatcherAssert.assertThat(
-            baos.toString(), Matchers.containsString("how are you")
+            baos.toString(), Matchers.containsString(EmAliasTest.EMAIL_BODY)
         );
     }
 
+    /**
+     * EmAlias does not send an email to the sender of the message.
+     * @throws Exception If there is some problem inside
+     */
+    @Test
+    public void doesNotSendEmailToSender() throws Exception {
+        final Postman postman = Mockito.mock(Postman.class);
+        final MkBase base = new MkBase();
+        final Alias alias = new EmAlias(base.randomAlias(), postman);
+        final Alias friend = base.randomAlias();
+        final Bout bout = alias.inbox().bout(alias.inbox().start());
+        bout.friends().invite(friend.name());
+        bout.friends().invite(alias.name());
+        bout.messages().post(EmAliasTest.EMAIL_BODY);
+        final ArgumentCaptor<Envelope> captor =
+            ArgumentCaptor.forClass(Envelope.class);
+        Mockito.verify(postman, Mockito.times(1)).send(captor.capture());
+        final Message msg = captor.getValue().unwrap();
+        MatcherAssert.assertThat(
+            msg.getAllRecipients().length,
+            Matchers.is(1)
+        );
+        final InternetAddress address =
+            (InternetAddress) msg.getAllRecipients()[0];
```

Comment:
@ndbroadbent here you are checking only the first recipient. We should check that none of the recipients equals the sender.